functionaltests/src/test/java/specs/response/JsonResponseFixture.java

```diff
@@ -0,0 +1,112 @@
+
+    public String createCardPayment(String paymentName,
+                                    String paymentMethod,
+                                    String transactionType,
+                                    String centAmount,
+                                    String currencyCode,
+                                    String languageCode) throws ExecutionException, InterruptedException, UnsupportedEncodingException {
+
+        return createAndSaveCardPayment(paymentName, paymentMethod, transactionType, centAmount, currencyCode, languageCode);
+    }
+
+    public String createWalletPayment(
+            final String paymentName,
+            final String paymentMethod,
+            final String transactionType,
+            final String centAmount,
+            final String currencyCode,
+            final String languageCode) throws ExecutionException, InterruptedException, UnsupportedEncodingException {
+
+        final MonetaryAmount monetaryAmount = createMonetaryAmountFromCent(Long.valueOf(centAmount), currencyCode);
+
+        final String successUrl = baseRedirectUrl + URLEncoder.encode(paymentName + " Success", "UTF-8");
+        final String errorUrl = baseRedirectUrl + URLEncoder.encode(paymentName + " Error", "UTF-8");
+        final String cancelUrl = baseRedirectUrl + URLEncoder.encode(paymentName + " Cancel", "UTF-8");
+        final PaymentDraft paymentDraft = PaymentDraftBuilder.of(monetaryAmount)
+                .paymentMethodInfo(PaymentMethodInfoBuilder.of()
+                        .method(paymentMethod)
+                        .paymentInterface("PAYONE")
+                        .build())
+                .custom(CustomFieldsDraft.ofTypeKeyAndObjects(
+                        CustomTypeBuilder.PAYMENT_WALLET,
+                        ImmutableMap.<String, Object>builder()
+                                .put(CustomFieldKeys.LANGUAGE_CODE_FIELD, languageCode)
+                                .put(CustomFieldKeys.SUCCESS_URL_FIELD, successUrl)
+                                .put(CustomFieldKeys.ERROR_URL_FIELD, errorUrl)
+                                .put(CustomFieldKeys.CANCEL_URL_FIELD, cancelUrl)
+                                .put(CustomFieldKeys.REFERENCE_FIELD, "<placeholder>")
+                                .build()))
+                .build();
+
+        Payment payment = createPaymentFromDraft(paymentName, paymentDraft, transactionType);
+        return payment.getId();
+    }
+
+    /**
+     * Returns {@link JsonNode} with key-value String map from Payone response.
+     * @param paymentName previously created payment name from HTML template
+     * @return complete {@link JsonNode} with expected fields from Payone API (errorcode, errormessage, customermessage)
+     * @throws ExecutionException
+     * @throws IOException
+     */
+    public JsonNode handleErrorJsonResponse(final String paymentName) throws ExecutionException, IOException {
```

Comment:
fixed in bf3d5ad4df68b365226146ed2754608357630054 and 418fbf79ce050b1d87ed919823195bc9504e2df3